source/extensions/filters/http/abac/abac_filter.cc

```diff
@@ -0,0 +1,171 @@
+#include "extensions/filters/http/abac/abac_filter.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ABACFilter {
+
+using CelValue = google::api::expr::runtime::CelValue;
+using CelMap = google::api::expr::runtime::CelMap;
+using CelList = google::api::expr::runtime::CelList;
+
+namespace {
+
+constexpr absl::string_view kMetadata = "metadata";
+constexpr absl::string_view kConnection = "connection";
+constexpr absl::string_view kRequestedServerName = "requested_server_name";
+constexpr absl::string_view kLocalAddress = "local_address";
+constexpr absl::string_view kRemoteAddress = "remote_address";
+constexpr absl::string_view kCertificatePresented = "certificate_presented";
+constexpr absl::string_view kSubjectLocalCertificate = "subject_local_certificate";
+constexpr absl::string_view kIssuerPeerCertificate = "issuer_peer_certificate";
+constexpr absl::string_view kSubjectPeerCertificate = "subject_peer_certificate";
+constexpr absl::string_view kTlsVersion = "tls_version";
+constexpr absl::string_view kRequest = "request";
+constexpr absl::string_view kPath = "path";
+constexpr absl::string_view kHost = "host";
+constexpr absl::string_view kScheme = "scheme";
+constexpr absl::string_view kMethod = "method";
+constexpr absl::string_view kReferer = "referer";
+constexpr absl::string_view kHeaders = "headers";
+constexpr absl::string_view kTime = "time";
+
+class ConnectionWrapper : public CelMap {
+public:
+  ConnectionWrapper(const StreamInfo::StreamInfo& info) : info_(info) {}
+
+  absl::optional<CelValue> operator[](CelValue key) const override {
+    if (!key.IsString()) {
+      return {};
+    }
+    auto value = key.StringOrDie().value();
+    if (value == kRequestedServerName) {
+      return CelValue::CreateString(info_.requestedServerName());
+    } else if (value == kLocalAddress) {
+      return CelValue::CreateString(info_.downstreamLocalAddress()->asString());
+    } else if (value == kRemoteAddress) {
+      return CelValue::CreateString(info_.downstreamRemoteAddress()->asString());
+    }
+
+    auto ssl_info = info_.downstreamSslConnection();
+    if (ssl_info != nullptr) {
+      if (value == kCertificatePresented) {
+        return CelValue::CreateBool(ssl_info->peerCertificatePresented());
+      } else if (value == kSubjectLocalCertificate) {
+        return CelValue::CreateString(ssl_info->subjectLocalCertificate());
```

Comment:
Note: I realize this is incorrect since string_view loses its parent string here. I think it would be good to refactor stream info to present more string_views.